packages/cms/src/components/fields/ProfileSearch.jsx

```diff
@@ -279,38 +282,52 @@ class ProfileSearch extends Component {
             (position !== "absolute" || active) && results
               ? (() => {
 
-                if (!results.grouped.length) {
+                if (mode === "profile" && !results.grouped.length || mode === "dimension" && !dimensionResults.length) {
                   return <NonIdealState key="empty" icon="zoom-out" title={`No results matching "${query}"`} />;
                 }
 
-                switch (display) {
-
-                  case "columns":
-                    const columnProfiles = Object.keys(results.profiles || {})
-                      .filter(d => !availableProfiles.length || availableProfiles.includes(d))
-                      .sort((a, b) => {
-                        const aIndex = columnOrder.includes(a) ? columnOrder.indexOf(a) : columnOrder.length + 1;
-                        const bIndex = columnOrder.includes(b) ? columnOrder.indexOf(b) : columnOrder.length + 1;
-                        return aIndex - bIndex;
-                      })
-                      .map(profile => results.profiles[profile] || []);
-                    return <ProfileColumns columnTitles={profileTitles} tileProps={{joiner}} data={columnProfiles} />;
-
-                  default:
-                    const listProfiles = (results.grouped || [])
-                      .filter(d => !availableProfiles.length || availableProfiles.includes(d[0].slug));
-                    return (
-                      <ul key="list" className="cms-profilesearch-list">
-                        {listProfiles.map((result, j) =>
-                          <li key={`r-${j}`} className="cms-profilesearch-list-item">
-                            <Link to={linkify(router, result, locale)} className="cms-profilesearch-list-item-link">
-                              {result.map(d => formatTitle(d.name)).join(` ${joiner} `)}
-                              <div className="cms-profilesearch-list-item-sub u-font-xs">{profileTitles[result.map(d => d.slug).join("/")] || formatCategory([result])}</div>
-                            </Link>
-                          </li>
-                        )}
-                      </ul>
-                    );
+                if (mode === "dimension") {
+                  return <ul key="list" className="cms-profilesearch-list">
+                    {dimensionResults.map((result, j) =>
+                      <li key={`r-${j}`} className="cms-profilesearch-list-item">
+                        <Link to={linkify(router, result, locale)} className="cms-profilesearch-list-item-link">
+                          {result.map(d => formatTitle(d.name)).join(` ${joiner} `)}
+                          <div className="cms-profilesearch-list-item-sub u-font-xs">{profileTitles[result.map(d => d.slug).join("/")] || formatCategory([result])}</div>
+                        </Link>
+                      </li>
+                    )}
+                  </ul>;
+                }
+                else {
```

Comment:
ehhhh I'm not a fan of duping this "list" formatting code. How would you feel about ditching the `mode` prop all together, and just adjusting the way you `formatResults` to include like, a dummy column?